### ISSUE TITLE
create replication vignette for svysen

### ISSUE THREAD

**Issue by ajdamico** [ajdamico/convey#212]

**Comment by guilhermejacob** [comment]
This is a measure that depends on the Gini index, which doesn't have a unique result (depends on how you calculate quantiles), and it isn't used very often.

An idea would be a decomposition: the Sen index can be written as a function of the FGT(0), FGT(1) and Gini among the poor (see https://github.com/DjalmaPessoa/convey/blob/master/R/svysen.R#L158).
If we prove that FGT(0), FGT(1) and Gini (among the poor) estimates are correct, it is easy to show what would be the Sen index. The variance, however, is a little bit more complicated, as it has to account for covariances.
Actually, this decomposition is done inside the `svysen` function and returned when `components = TRUE`.

I can do this explicitly if you agree, @DjalmaPessoa and @ajdamico.

**Comment by guilhermejacob** [comment]
Building it from components:
```
# load the convey package
library(convey)

# load the survey library
library(survey)

# load the vardpoor library
library(vardpoor)

# load the synthetic european union statistics on income & living conditions
data(eusilc)

# make all column names lowercase
names( eusilc ) <- tolower( names( eusilc ) )

# construct a survey.design
# using our recommended setup
des_eusilc <- 
  svydesign( 
    ids = ~ rb030 , 
    strata = ~ db040 ,  
    weights = ~ rb050 , 
    data = eusilc
  )

# immediately run the convey_prep function on it
des_eusilc <- convey_prep( des_eusilc )

# FGT0
fgt0 <- svyfgt( ~eqincome , des_eusilc , abs_thresh = 10000 , g = 0 )

# FGT1
fgt1 <- svyfgt( ~eqincome , des_eusilc , abs_thresh = 10000 , g = 1 )

# Gini of poor incomes
gini <- svygini( ~eqincome , subset( des_eusilc , eqincome < 10000 ) )

# Sen Index
sen <- svysen( ~eqincome , des_eusilc , abs_thresh = 10000 , components = TRUE )
sen.comp <- attr( sen , "components" )

# Decomposing the Sen index
sen.comp$Components[1,][[1]] == coef(fgt0)[[1]]
sen.comp$Components[2,][[1]] == coef(fgt1)[[1]]
sen.comp$Components[3,][[1]] == coef(gini)[[1]]

# load the convey package
library(convey)

# load the survey library
library(survey)

# load the vardpoor library
library(vardpoor)

# load the synthetic european union statistics on income & living conditions
data(eusilc)

# make all column names lowercase
names( eusilc ) <- tolower( names( eusilc ) )

# construct a survey.design
# using our recommended setup
des_eusilc <- 
  svydesign( 
    ids = ~ rb030 , 
    strata = ~ db040 ,  
    weights = ~ rb050 , 
    data = eusilc
  )

# immediately run the convey_prep function on it
des_eusilc <- convey_prep( des_eusilc )

# FGT0
fgt0 <- svyfgt( ~eqincome , des_eusilc , abs_thresh = 10000 , g = 0 )

# FGT1
fgt1 <- svyfgt( ~eqincome , des_eusilc , abs_thresh = 10000 , g = 1 )

# Gini of poor incomes
gini <- svygini( ~eqincome , subset( des_eusilc , eqincome < 10000 ) )

# Sen Index
sen <- svysen( ~eqincome , des_eusilc , abs_thresh = 10000 , components = TRUE )
sen.comp <- attr( sen , "components" )

# Decomposing the Sen index
sen.comp$Components[1,][[1]] == fgt0[[1]]
sen.comp$Components[2,][[1]] == fgt1[[1]]
sen.comp$Components[3,][[1]] == gini[[1]]

# calculate from components
fgt0[[1]] * gini[[1]] + fgt1[[1]] * ( 1 - gini[[1]] ) == sen[[1]]
```

This shows how the Sen index can be decomposed.

**Comment by ajdamico** [comment]
is it possible (and of any value) to match some measure of variance?  it looks like these calculations only build the components for the coefficients..  sorry if i'm misunderstanding something and thank you for all this awesome work

**Comment by guilhermejacob** [comment]
916e38a8310d022af81b8de93eb8568689e41764 sets svysen as experimental

**Comment by guilhermejacob** [comment]
closed by 984351c8f2bd275a76b453507ba87854fdd2db59
